entity-service/src/main/kotlin/com/egm/stellio/entity/repository/QueryUtils.kt

```diff
@@ -201,4 +225,24 @@ object QueryUtils {
                }
             """.trimIndent()
         }
+
+    private fun buildGeoQueryFilter(geoQuery: GeoQuery, prefix: String = "AND"): String {
+        val coordinates = geoQuery.coordinates.toString().split("[\\[,\\]]".toRegex())
+        val georelParams = extractGeorelParams(geoQuery.georel!!)
+        return """
+                $prefix entity.location CONTAINS '${geoQuery.geometry!!.uppercase()}'
+                AND point.distance(
+                     point({
+                        latitude: toFloat(apoc.text.regexGroups(entity.location, '([\\d\\.]+) ([\\d\\.]+)')[0][1]), 
+                        longitude: toFloat(apoc.text.regexGroups(entity.location, '([\\d\\.]+) ([\\d\\.]+)')[0][2]),
+                        crs: 'wgs-84'
+                     }),
+                     point({
+                        latitude: ${coordinates[1].toFloat()}, 
+                        longitude: ${coordinates[2].toFloat()}, 
+                        crs: 'wgs-84'
+                     })
```

Comment:
If I refer to https://www.rfc-editor.org/rfc/rfc7946#section-3.1.1, first element is longitude, and second is latitude. Seems you are doing the opposite.

Reply:
Yes indeed I change it